### PR TITLE
Implement feedback2

### PR DESCRIPTION
**Pull request by andela-hngerebara**
**What does this PR do?**
Implement feedback received

**Description of Task to be completed?**
* Remove irrelevant fields like password, createdAt, updatedAt from being returned
* Use appropriate status code responses and remove additional message being returned for valid responses
* Destructure request in controllers 
* Fix API Documentation responses (swagger) responses
* Restrict user from creating documents with empty fields( all fields must be entered)
* Allow users to update document access type (example from private to public)
* Admin to successfully delete a user without reloading the page before user deleted is updated
* Display message for successful delete of a user
* Remove pagination from users when search is not found
* Refactor documents model to have varying character limits based on the fields
*Add PR style guide for contributors
* Add Acceptance criteria to all stories on Pivotal tracker
* Fix all sequelize validation errors
* Fix users profile to be modified without reloading the page
* Clear console errors
* Create a webpack for production that minifies static pages
* Fix all failing tests
* Correct test suite descriptions to match what is being tested for
* Add proper tests rather than test for just keys.
* Fix all variables that do not adhere to the style guide
* JSDOC to be properly described
* Add code climate badge
* Fix components that have .js extension

**How should this be manually tested?**
All of the aforementioned can be tested [here](https://hopeazdms-dev.herokuapp.com)

**What are the relevant pivotal tracker stories?**
N/A

### CODE REVIEW

File: client/src/components/users/manageUsers/UsersActions.js [hunk 1]
```diff
@@ -82,28 +86,27 @@ export const updateUserSuccess = data => ({
...
 
 export const updateFailureMessage = errorMessage => ({
   type: types.UPDATE_FAILURE_MESSAGE,
   errorMessage
 });
 
-export const updateUserProfile = updatedUser => (dispatch) => {
-  return axios.put(`/users/${updatedUser.id}`, updatedUser)
-    .then((response) => {
+export const updateUserProfile = updatedUser => (dispatch) => axios.put(`/users/${updatedUser.id}`, updatedUser)
```
Comment (temilaj) [inline client/src/components/users/manageUsers/UsersActions.js]: It's great practice to omit the braces If your function takes a single argument.

File: server/controllers/documents.js [hunk 2]
```diff
@@ -119,9 +119,11 @@ const documentsController = {
           .then(() => res.status(200).send({
             message: 'Document deleted successfully.'
           }))
-          .catch(error => res.status(400).send(error));
+          .catch(() => res.status(409).send({
+            message: 'Could not delete document'
+          }));
       })
-      .catch(error => res.status(500).send(error));
+      .catch(error => res.status(400).send(error));
```
Comment (temilaj) [inline server/controllers/documents.js]: I see you took out the 500 status code for a 400. Why is that?

File: tests/e2e/DocumentTest.spec.js [hunk 3]
```diff
@@ -1,35 +1,39 @@
...
-      .pause(2000)
-      .waitForElementVisible('#search-btn', 1000)
-      .click('#search-btn')
-      .pause(1000);
+      .setValue('input[id=search-input]', 'hello')
+      .pause(5000)
+      .waitForElementVisible('.card.small', 3000)
+      .assert.elementPresent('.card.small')
+      .waitForElementVisible('#search-btn', 3000)
+      .click('#search-btn');
```
Comment (temilaj) [inline tests/e2e/DocumentTest.spec.js]: It seems to me like this test is testing that we can click on a search button. as opposed to testing for what the description says it's testing for.

File: tests/e2e/DocumentTest.spec.js [hunk 4]
```diff
@@ -39,44 +43,48 @@ module.exports = {
...
+      .waitForElementVisible('iframe', 3000)
       .frame(0)
-       .setValue('.mce-content-body', 'dfdfdfdffdfdfdfdfsfdsfsfsdfsdfdfdfdfdfdfdfdfdfdf')
-       .frame(null)
+      .setValue('.mce-content-body', 'dfdfdfdffdfdfdfdfsfdsfsfsdfsdfdfdfdfdfdfdfdfdfdf')
+      .frame(null)
       .click('button[type=submit]')
-      .pause(2000);
+      .waitForElementVisible('.toast', 3000)
+      .assert.containsText('.toast', 'Document saved');
```
Comment (temilaj) [inline tests/e2e/DocumentTest.spec.js]: How are we sure that the document was truly created if we can assert by the notification. Can you think about something more to test for here ?

File: tests/e2e/EditProfile.Test.spec.js [hunk 5]
```diff
@@ -16,9 +17,11 @@ module.exports = {
       .click('#edit-profile')
       .setValue('input[name="firstName"]', ' Editted')
       .setValue('input[name="lastName"]', ' Editted')
-      .assert.value('input[name="firstName"]', 'Hope Editted')
-      .assert.value('input[name="lastName"]', 'Ngerebara Editted')
+      .assert.value('input[name="firstName"]', 'admin2 Editted')
+      .assert.value('input[name="lastName"]', 'admin Editted')
       .click('button[id=edit]')
+      .waitForElementVisible('.toast', 3000)
+      .assert.containsText('.toast', 'Profile updated successfully')
```
Comment (temilaj) [inline tests/e2e/EditProfile.Test.spec.js]: are we really testing that users can edit their profiles, or are we testing that the toast message is sent after a button is clicked?

File: tests/e2e/UsersTest.spec.js [hunk 6]
```diff
@@ -10,28 +10,26 @@ module.exports = {
...
-  'User should be able to view all users': (browser) => {
+  'Admin should be able to view all users': (browser) => {
     browser
     .resizeWindow(1280, 800)
     .click('#view-users')
       .waitForElementVisible('#users-list', 3000)
-      .assert.urlEquals('http://localhost:8080/users')
-      .pause(2000);
+      .assert.elementPresent('ul #single-user')
+      .assert.urlEquals('http://localhost:8080/users');
```
Comment (temilaj) [inline tests/e2e/UsersTest.spec.js]: please see me as regards your tests

File: tests/e2e/DocumentTest.spec.js (hunk 3 above)
Comment (andela-hngerebara) [inline tests/e2e/DocumentTest.spec.js]: This test checks if the card(document) displays when the search begins.

File: client/src/components/auth/AuthActions.js [hunk 7]
```diff
@@ -26,14 +26,13 @@ const checkinUserAction = ({ email, password }) => dispatch =>
   axios
     .post('/users/login', { email, password })
     .then((response) => {
-      const message = response.data.message;
       const token = response.data.token;
       localStorage.setItem('token', token);
       const payload = API_URL === '9999'
```
Comment (temilaj) [inline client/src/components/auth/AuthActions.js]: what does this line of code do?

File: tests/e2e/DocumentTest.spec.js (hunk 4 above)
Comment (andela-hngerebara) [inline tests/e2e/DocumentTest.spec.js]: Yes.
 I have included a test to check that particular document has been added to the list

File: tests/e2e/UsersTest.spec.js (hunk 6 above)
Comment (andela-hngerebara) [inline tests/e2e/UsersTest.spec.js]: Okay

File: client/src/components/auth/AuthActions.js (hunk 7 above)
Comment (andela-hngerebara) [inline client/src/components/auth/AuthActions.js]: I use it to separate  my code for testing purposes. 
I declared my API_URL globally therefore each time my test run, it picks the URL for development and also tries decoding the token which for my tests won't work. I had to declare a separate API_URL in my jest config.js.
This line of code checks if the API_URL set is for the test environment or not. If it is, it sends the email and password I provided for testing else it proceeds with decoding the token gotten  to be further used in my application.

File: tests/e2e/EditProfile.Test.spec.js (hunk 5 above)
Comment (andela-hngerebara) [inline tests/e2e/EditProfile.Test.spec.js]: Fixed it here https://github.com/andela-hngerebara/Document-Management-System/pull/88

File: client/src/components/users/manageUsers/UsersActions.js (hunk 1 above)
Comment (andela-hngerebara) [inline client/src/components/users/manageUsers/UsersActions.js]: Okay

**Pull request merged**